### PR TITLE
fix/request-reschedule-google-calendar-sync

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Add required param for deleting event on G-calendar.

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Now while request rescheduling g calendar event should be deleted.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/pages/api/book/request-reschedule.ts [hunk 1]
```diff
@@ -193,38 +193,17 @@ const handler = async (
...
-          );
-          const attendeeData = await findUserDataByUserId(destinationCalendar.userId);
-          const attendeeCredentialsMap = new Map();
-          attendeeData.credentials.forEach((credential) => {
-            attendeeCredentialsMap.set(credential.type, credential);
-          });
-          bookingRefsFiltered.forEach((bookingRef) => {
-            if (bookingRef.uid) {
-              const calendar = getCalendar(attendeeCredentialsMap.get(destinationCalendar.integration));
-              calendar?.deleteEvent(bookingRef.uid, builder.calendarEvent);
```
Comment (alannnc) [inline apps/web/pages/api/book/request-reschedule.ts:222]: This code was trying to delete an already deleted event from above.

File: apps/web/pages/api/book/request-reschedule.ts [hunk 2]
```diff
@@ -193,38 +193,17 @@ const handler = async (
           if (bookingRef.type.endsWith("_calendar")) {
             const calendar = getCalendar(credentialsMap.get(bookingRef.type));
 
-            return calendar?.deleteEvent(bookingRef.uid, builder.calendarEvent);
+            return calendar?.deleteEvent(
+              bookingRef.uid,
+              builder.calendarEvent,
+              bookingRef.externalCalendarId
```
Comment (alannnc) [inline apps/web/pages/api/book/request-reschedule.ts:199]: A new parameter was added to deleteEvent, but it wasn’t being send from here.

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 3]
```diff
@@ -224,20 +224,28 @@ export default class GoogleCalendarService implements Calendar {
         version: "v3",
         auth: myGoogleAuth,
       });
+      // Defaulting to primary calendar to avoid error from google api
+      let calendarId = event.destinationCalendar?.externalId ?? "primary";
```
Comment (alannnc) [inline packages/app-store/googlecalendar/lib/CalendarService.ts:228]: We had a conditional but wasn't preventing from sending null to google. We now at least try to delete on primary calendar from the user if it exist.

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 4]
```diff
@@ -224,20 +224,28 @@ export default class GoogleCalendarService implements Calendar {
...
-            /* 410 is when an event is already deleted on the Google cal before on cal.com
-            404 is when the event is on a different calendar */
+            /**
+             *  410 is when an event is already deleted on the Google cal before on cal.com
+             *  404 is when the event is on a different calendar
+             */
+            if (err.code === 410) return resolve();
             console.error("There was an error contacting google calendar service: ", err);
-            if (err.code === 410 || err.code === 404) return resolve();
+            if (err.code === 404) return resolve();
```
Comment (alannnc) [inline packages/app-store/googlecalendar/lib/CalendarService.ts:248]: 410 shouldn’t log anything, probably user deleted event on his calendar, and we're trying to delete it.

**Comment by hariombalhara** [comment]
Can confirm that it fixes the issue.

## Google calendar  and Apple Calendar Test
**Before**
- Booked an event
- Requested reschedule. It cancelled the event in cal.com but It remains in google calendar. Received emails.

**After**
- Booked an event 
- Requested reschedule. It cancelled the event in cal.com and in google calendar. Received emails.Received emails

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Though it fixes the reported issue. Left some comments to be sure we are handling all cases and things are documented.

**Comment by PeerRich** [comment]
@zomars do we wait for this to be addressed and merged before 1.7.3?

File: apps/web/pages/api/book/request-reschedule.ts [hunk 5]
```diff
@@ -193,38 +193,17 @@ const handler = async (
           if (bookingRef.type.endsWith("_calendar")) {
             const calendar = getCalendar(credentialsMap.get(bookingRef.type));
 
-            return calendar?.deleteEvent(bookingRef.uid, builder.calendarEvent);
+            return calendar?.deleteEvent(
```
[the remark this answers is not in the document]
Comment (emrysal) [reply apps/web/pages/api/book/request-reschedule.ts:196]: Other calendar integrations might not have a concept of externalCalendarId - or just don't support this yet. Worth looking into but the key thing to keep in mind is that this comes down to a shared interface.

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 6]
```diff
@@ -224,20 +224,28 @@ export default class GoogleCalendarService implements Calendar {
         version: "v3",
         auth: myGoogleAuth,
       });
+      // Defaulting to primary calendar to avoid error from google api
+      let calendarId = event.destinationCalendar?.externalId ?? "primary";
+      if (externalCalendarId) {
```
[the remark this answers is not in the document]
Comment (emrysal) [reply packages/app-store/googlecalendar/lib/CalendarService.ts:229]: I think externalCalendarId not being ?'d is an error. Or it is expected that the "primary" fallback is provided manually.

**Review by emrysal — APPROVED** [review]

**Comment by emrysal** [comment]
This is safe to go in as is @PeerRich - but we should indeed look into the flows used for externalCalendarId's and also how this works for other integrations.

File: packages/app-store/googlecalendar/lib/CalendarService.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (alannnc) [reply packages/app-store/googlecalendar/lib/CalendarService.ts:229]: You were both right, I've updated accordingly to be more clear about what was needed.

- google api was receiving calendarId: null thus throwing error.
- added default in a case when calendarId could receive a null value.

File: apps/web/pages/api/book/request-reschedule.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (alannnc) [reply apps/web/pages/api/book/request-reschedule.ts:196]: Yes, exactly what @emrysal said.

[the remark this answers is not in the document]
Comment (hariombalhara) [reply apps/web/pages/api/book/request-reschedule.ts:196]: Cool, we should then work on having different function signatures in interface for two different cases

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**